references/cli/livediff.go

```diff
@@ -0,0 +1,117 @@
+	cmdutil "github.com/oam-dev/kubevela/pkg/utils/util"
+	"github.com/oam-dev/kubevela/references/appfile/dryrun"
+)
+
+type livediffCmdOptions struct {
+	dryRunCmdOptions
+	revision string
+	context  int
+}
+
+// NewLiveDiffCommand creates `live-diff` command
+func NewLiveDiffCommand(c common.Args, ioStreams cmdutil.IOStreams) *cobra.Command {
+	o := &livediffCmdOptions{
+		dryRunCmdOptions: dryRunCmdOptions{
+			IOStreams: ioStreams,
+		}}
+	cmd := &cobra.Command{
+		Use:                   "live-diff",
+		DisableFlagsInUseLine: true,
+		Short:                 "Dry-run an application, and do diff on a specific app revison",
+		Long:                  "Dry-run an application, and do diff on a specific app revison. The provided capability definitions will be used during Dry-run. If any capabilities used in the app are not found in the provided ones, it will try to find from cluster.",
+		Example:               "vela live-diff -f app-v2.yaml -r app-v1 --context 10",
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			return c.SetConfig()
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			newClient, err := c.GetClient()
+			if err != nil {
+				return err
+			}
+
+			velaEnv, err := GetEnv(cmd)
+			if err != nil {
+				return err
+			}
+			objs := []oam.Object{}
+			if o.definitionFile != "" {
+				objs, err = ReadObjectsFromFile(o.definitionFile)
+				if err != nil {
+					return err
+				}
+			}
+			pd, err := c.GetPackageDiscover()
+			if err != nil {
+				return err
+			}
+
+			dm, err := discoverymapper.New(c.Config)
+			if err != nil {
+				return err
+			}
+
+			app, err := readApplicationFromFile(o.applicationFile)
+			if err != nil {
+				return errors.WithMessagef(err, "read application file: %s", o.applicationFile)
+			}
+			if app.Namespace == "" {
+				app.SetNamespace(velaEnv.Namespace)
+			}
+
+			appRevision := &v1beta1.ApplicationRevision{}
+			if err := newClient.Get(context.Background(),
+				client.ObjectKey{Name: o.revision, Namespace: app.Namespace}, appRevision); err != nil {
+				return errors.Wrapf(err, "cannot get AppRevision %q", o.revision)
+			}
+
+			liveDiffOption := dryrun.NewLiveDiffOption(newClient, dm, pd, o.revision, objs)
+			diffResult, err := liveDiffOption.Diff(context.Background(), app, appRevision)
+			if err != nil {
+				return errors.WithMessage(err, "cannot calculate diff")
+			}
+
+			var buff = bytes.Buffer{}
+			reportDiffOpt := dryrun.NewReportDiffOption(o.context, &buff)
+			reportDiffOpt.PrintDiffReport(diffResult)
+			o.Info(buff.String())
+			return nil
+		},
+	}
+
+	cmd.Flags().StringVarP(&o.applicationFile, "file", "f", "./app.yaml", "application file name")
+	cmd.Flags().StringVarP(&o.definitionFile, "definition", "d", "", "specify a definition file or directory, it will only be used in dry-run rather than applied to K8s cluster")
+	cmd.Flags().StringVarP(&o.revision, "revision", "r", "", "specify an application revision name")
```

Comment:
```suggestion
	cmd.Flags().StringVarP(&o.revision, "revision", "r", "", "specify an application revision name, by default, it will compare with the latest revision")
```